scio-test/src/main/scala/com/spotify/scio/testing/JobTest.scala

```diff
@@ -65,26 +65,32 @@ import scala.util.control.NonFatal
  */
 object JobTest {
 
-  case class Builder(className: String, cmdlineArgs: Array[String],
+  private case class BuilderState(className: String, cmdlineArgs: Array[String],
                      inputs: Map[TestIO[_], Iterable[_]],
                      outputs: Map[TestIO[_], SCollection[_] => Unit],
-                     distCaches: Map[DistCacheIO[_], _]) {
+                     distCaches: Map[DistCacheIO[_], _], wasRunInvoked: Boolean = false)
```

Comment:
Why adding this class? To make internal state like `inputs`, `outputs` invisible to users?
Also align parameters?